src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/TelemetryHttpModule.cs

```diff
@@ -85,7 +85,7 @@ private void OnExecuteRequestStep(HttpContextBase context, Action step)
     {
         // Called only on 4.7.1+ runtimes
 
-        if (context.CurrentNotification == RequestNotification.ExecuteRequestHandler && !context.IsPostNotification)
+        if (!context.IsPostNotification)
```

Comment:
Let's remove this check as well and call `ActivityHelper.RestoreContextIfNeeded(context.ApplicationInstance.Context)` on every step.

I tested it locally and with this check in place `PostRequestHandlerExecute` won't be restored.